src/rubintv/handlers/external/endpoints.py

```diff
@@ -196,12 +201,18 @@ async def update_todays_table(request: web.Request) -> web.Response:
             bucket.name, camera.slug, date_str, logger
         )
 
+        per_day = {}
+        movie = get_movie_url(bucket, camera, the_date, logger)
+        if movie:
+            per_day["movie"] = movie
```

Comment:
Looks like `get_movie_url()` returns either an empty string (unless the request gets a 200) or the URL, so I think this should be fine.

Reply:
Then I would leave out the check and let an exception raise.  This really looks like the check is trying to do something

Reply:
As discussed OOB, the check is (I think, @ugyballoons please chime in here) trying to make sure that they key is only in the dict in the case that the movie URL points to an existing movie.

However, as also discussed OOB @ugyballoons, Simon doesn't think that this model of using the presence/absence of keys in the dict as control flow over whether to display the movie icon is a good plan (assuming that I'm right that that's what's being done). So maybe do comment on this as to how this is intended to work, and if I'm right about how that is, we can come up with a more robust plan here.

Reply:
OK.  Fair enough.  There is no need to re-engineer the entire internals for this, but it would be worth thinking about in any upcoming rewrite.  As it stands, it is pretty confusing to read.

Reply:
Ah, OK, thanks. In that case, Guy, if I'm right, could you just add a code comment about the presence of that key being control for generating items on the page later, and make a note to self for future redesign perhaps? And if I've got it wrong, just clarify what's happening here for us both?

Reply:
(also, side note to Guy - if you don't know about the new "walrus operator", this is a perfect example to write:
```if movie := get_movie_url(bucket, camera, the_date, logger)
    per_day["movie"] = movie
```
 🙂)

Reply:
& hello Simon! I don't think I said hello yet.

I didn't know about the the walrus. Looks good!
The intention behind the dict is as you suspect Merlin, if it's empty, the view can just iterate through and display nothing if there's nothing there. The idea was to keep as much logic in the controller as poss. What's a more satisfactory way of dealing with this?